examples/authorized_api/authorization.go

```diff
@@ -3,10 +3,59 @@ package main
 import (
 	"encoding/json"
 	"errors"
+	"fmt"
 	"io/ioutil"
 	"net/http"
 )
 
+func ZebedeeMustPermitUser(handler http.HandlerFunc) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		status, err := zebedeeValidateUser()
+		if err != nil {
+			if status == "401 Unauthorized" {
+				http.Error(w, err.Error(), http.StatusUnauthorized)
+			} else {
+				http.Error(w, err.Error(), http.StatusInternalServerError)
+			}
+			return
+		} else {
+			w.WriteHeader(200)
+			message := "[\"DELETE\", \"READ\", \"CREATE\", \"UPDATE\"]"
+			fmt.Fprintf(w, message)
+		}
+		handler(w, r)
+	}
+}
+
+func zebedeeValidateUser() (string, error) {
+	type Permissions struct {
+		PermissionsMsg string `bson:"message" json:"message"`
+	}
+	var permissions Permissions
+	config := NewConfig()
+	response, err := http.Get(config.authorizationServiceUrl + "/userInstancePermissions")
+	status := response.Status
```

Comment:
Im not 100% sure on this as the `http.Get` documentation doesn't cover it but if `err != nil` then the response var could be `nil` which would cause a nil pointer err when `status := response.Status` is called.

To be on the safe side I would suggest handling the error first before doing anything with the response i.e.:
````
	response, err := http.Get(config.authorizationServiceUrl + "/userInstancePermissions")
	if err != nil {
		return "", err
	}

	status := response.Status
	if status == "401 Unauthorized" {
        }
        ....
````
That should protect us against any unexpected error.

Reply:
Also just noticed this - it's probably better to use `response.StatusCode` instead of `response.Status` (status code will return an int which IMO is better to operate on than a string - the naming of the fields is confusing.

Reply:
Good spot about the error, and thanks for the info about StatusCode (I think I should have known that already) - done